code/VideoGroup.php

```diff
@@ -37,138 +37,38 @@ class VideoGroup extends Page
      */
     private static $description = 'Videos Landing Page';
 
-
-    /**
-     * loadDescendantVideoGroupIDListInto function.
-     *
-     * @access public
-     * @param mixed &$idList
-     * @return void
-     */
-    public function loadDescendantVideoGroupIDListInto(&$idList)
-    {
-        if ($children = $this->AllChildren()) {
-            foreach ($children as $child) {
-                if (in_array($child->ID, $idList)) {
-                    continue;
-                }
-
-                if ($child instanceof VideoGroup) {
-                    $idList[] = $child->ID;
-                    $child->loadDescendantVideoGroupIDListInto($idList);
-                }
-            }
-        }
-    }
-
-
     /**
      * VideoGroupIDs function.
      *
      * @access public
-     * @return void
+     * @return array
      */
     public function VideoGroupIDs()
     {
-        $holderIDs = array();
-        $this->loadDescendantVideoGroupIDListInto($holderIDs);
-        return $holderIDs;
+        return $this->AllChildren()->filter('ClassName', 'VideoGroup')->column('ID');
     }
 
     /**
      * Videos function.
      *
      * @access public
-     * @return PaginatedList
+     * @return DataList
      */
     public function getVideoList()
     {
-
-        $filter = '"ParentID" = ' . $this->ID;
-        $limit = 3;
-
-        // Build a list of all IDs for VideoGroups that are children
-        $holderIDs = $this->VideoGroupIDs();
-
-        if ($holderIDs) {
-            if ($filter) {
-                $filter .= ' OR ';
-            }
-            $filter .= '"ParentID" IN (' . implode(',', $holderIDs) . ")";
-        }
-
-        $order = '"SiteTree"."Title" ASC';
-
-        $entries = Video::get()->where($filter)->sort($order);
-
-        $list = new PaginatedList($entries, Controller::curr()->request);
-        $list->setPageLength($limit);
-        return $list;
+        $groupIDS = $this->VideoGroupIDS();
+        $groupIDS[] = $this->ID;
+        return Video::get()->filter('ParentID', $groupIDS)->sort(array('Title' => 'ASC'));
```

Comment:
That might be better, I was going off of the existing sort. I'll update it to be off of the sort column. Maybe we add a config option later to allow for custom sorting.